### PR TITLE
ERC721 related cli CMDs

### PR DESCRIPTION
**Pull request by MakMuftic**
## Description
ERC721 and ERC721 handler deploy and interaction CLI commands

## Related Issue Or Context

Closes: #95 

## How Has This Been Tested? Testing details.
Tested all commands on chainbridge-core-example repo manually.
Will expand testing with next PR for issue #96 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Documentation

## Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [X] I have updated the documentation locally and in chainbridge-docs (**only locally**)
- [ ] I have added tests to cover my changes.
- [ ] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: chains/evm/cli/erc721/add-minter.go [hunk 1]
```diff
@@ -9,52 +18,73 @@ var addMinterCmd = &cobra.Command{
...
-	minter := cctx.String("minter")
-	if !common.IsHexAddress(minter) {
-		return errors.New("invalid minter address")
-	}
-	minterAddress := common.HexToAddress(minter)
-	ethClient, err := client.NewClient(url, false, sender, big.NewInt(0).SetUint64(gasLimit), big.NewInt(0).SetUint64(gasPrice), big.NewFloat(1))
+	gasPricer.SetClient(ethClient)
+	gasPricer.SetOpts(&evmgaspricer.GasPricerOpts{UpperLimitFeePerGas: gasPrice})
+
+	addMinterInput, err := calls.PrepareErc721AddMinterInput(ethClient, erc721Addr, minterAddr)
```
[the remark this answers is not in the document]
Comment (MakMuftic) [reply chains/evm/cli/erc721/add-minter.go:72]: I will refactor all calls not to use Prepare functions, but will pass entire `ethClient` as I think `Transactor` refactoring should be made in separate PR when resolving #124

File: chains/evm/calls/erc721.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import (
+	"math/big"
+	"strings"
+
+	"github.com/ChainSafe/chainbridge-core/chains/evm/calls/consts"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/rs/zerolog/log"
+)
+
```
[the remark this answers is not in the document]
Comment (MakMuftic) [reply chains/evm/calls/erc721.go:15]: I will wait for #124 to be resolved, as it doesn't make sense to write something that will be refactored as soon as this is resolved. This doesn't block this PR from being merged.

File: chains/evm/calls/deploy.go [hunk 3]
```diff
@@ -51,6 +51,31 @@ func DeployErc20Handler(c ClientDeployer, txFabric TxFabric, gasPriceClient GasP
...
+	return address, nil
+}
+
+func DeployErc721Handler(c ClientDeployer, txFabric TxFabric, gasPriceClient GasPricer, bridgeAddress common.Address) (common.Address, error) {
+	log.Debug().Msgf("Deployng ERC721 Handler with params: %s", bridgeAddress.String())
+	parsed, err := abi.JSON(strings.NewReader(consts.ERC721HandlerABI))
+	if err != nil {
+		return common.Address{}, err
+	}
+	address, err := deployContract(c, parsed, common.FromHex(consts.ERC721HandlerBin), txFabric, gasPriceClient, bridgeAddress) // TODO: add arguments
```
Comment (mpetrun5) [inline chains/evm/calls/deploy.go:72]: Why the TODO ?

File: chains/evm/calls/erc721.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+	}
+	input, err := a.Pack(method, args...)
+	if err != nil {
+		return a, []byte{}, err
+	}
+	return a, input, nil
+}
+
+func prepareERC721MintTokensInput(destAddr common.Address, tokenId *big.Int, metadata []byte) ([]byte, error) {
+	log.Debug().Msgf("")
```
Comment (mpetrun5) [inline chains/evm/calls/erc721.go:97]: ?

File: chains/evm/cli/deploy/deploy.go [hunk 5]
```diff
@@ -180,15 +199,40 @@ func DeployCLI(cmd *cobra.Command, args []string, txFabric calls.TxFabric, gasPr
...
-				log.Error().Err(ErrErc20TokenAndSymbolNotProvided)
-				return ErrErc20TokenAndSymbolNotProvided
+
+			deployedContracts["erc721Token"] = erc721Addr.String()
+
+		case "erc721Handler":
+			log.Debug().Msgf("deploying ERC721 handler..")
+
+			emptyAddress := common.Address{}
+			if bridgeAddr == emptyAddress {
```
Comment (mpetrun5) [inline chains/evm/cli/deploy/deploy.go:223]: Might as well have it just if bridgeAddr = common.Address{} if it is not used anywhere else.

File: chains/evm/cli/erc721/mint.go [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+	if err != nil {
+		log.Error().Err(fmt.Errorf("eth client intialization error: %v", err))
+		return err
+	}
+
+	gasPricer.SetClient(ethClient)
+	gasPricer.SetOpts(&evmgaspricer.GasPricerOpts{UpperLimitFeePerGas: gasPrice})
+
+	_, err = calls.ERC721Mint(ethClient, txFabric, gasPricer.(calls.GasPricer), gasLimit, tokenId, metadata, erc721Addr, dstAddress)
+	if err == nil {
```
Comment (mpetrun5) [inline chains/evm/cli/erc721/mint.go:94]: Can in be rather if err != nil { return err } and then log in happy case?
Would make more sense to always have not happy case return err.
There is a couple like this.

**Review by mpetrun5 — CHANGES_REQUESTED** [review]

File: chains/evm/cli/deploy/deploy.go (hunk 5 above)
Comment (MakMuftic) [reply chains/evm/cli/deploy/deploy.go]: This is something that was just waiting for Tihomirs changes in `deploy.go` to be merged as it is obsolete code with new CLI code structure.

File: chains/evm/cli/erc721/deposit.go [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+	txHash, err := calls.ERC721Deposit(
+		ethClient, txFabric, gasPricer.(calls.GasPricer), gasLimit, tokenId, destinationID, resourceId, bridgeAddr, recipientAddr,
+	)
+	if err != nil {
+		return err
+	}
+
+	log.Debug().Msgf("erc721 deposit hash: %s", txHash.Hex())
+	log.Info().Msgf("%s token were transferred to %s from %s", tokenId.String(), recipientAddr.Hex(), senderKeyPair.CommonAddress().String())
```
[the remark this answers is not in the document]
Comment (MakMuftic) [reply chains/evm/cli/erc721/deposit.go:99]: I will do the same on the ERC20 deposit :+1:

File: chains/evm/cli/deploy/deploy.go [hunk 8]
```diff
@@ -21,17 +22,22 @@ import (
 var ErrNoDeploymentFlagsProvided = errors.New("provide at least one deployment flag. For help use --help")
 var ErrErc20TokenAndSymbolNotProvided = errors.New("erc20Name and erc20Symbol flags should be provided")
 
+// TODO: Find a smarter way of handling this error creation
```
[the remark this answers is not in the document]
Comment (MakMuftic) [reply chains/evm/cli/deploy/deploy.go:25]: Actually, there is no need for this anymore as we mark all necessary flags as required, so I will just remove it.

**Review by P1sar — APPROVED** [review]

**Review by mpetrun5 — APPROVED** [review]

**Pull request merged**